yt_dlp/downloader/external.py

```diff
@@ -481,13 +482,16 @@ def _call_downloader(self, tmpfilename, info_dict):
 
         args = [encodeArgument(opt) for opt in args]
         args.append(encodeFilename(ffpp._ffmpeg_filename_argument(tmpfilename), True))
+        args.extend(['-progress', 'pipe:1', '-stats_period', '0.1'])
         self._debug_cmd(args)
 
-        proc = Popen(args, stdin=subprocess.PIPE, env=env)
+        proc = Popen(args, env=env, universal_newlines=True, encoding='utf8', stdin=subprocess.PIPE, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
         if url in ('-', 'pipe:'):
             self.on_process_started(proc, proc.stdin)
         try:
-            retval = proc.wait()
+            ffmpeg_progress_tracker = FFmpegProgressTracker(info_dict, args, proc, self._hook_progress, 'downloading', 'downloaded_bytes')
```

Comment:
Since all the details needed for starting the process is being passed to FFmpegProgressTracker, why not let it handle that as well. So something like this:

```py
        ffmpeg_progress_tracker = FFmpegProgressTracker(...)
        if url in ('-', 'pipe:'):
            proc = ffmpeg_progress_tracker.process
            self.on_process_started(proc, proc.stdin)
```